packages/grafana-ui/src/components/Icon/Icon.tsx

```diff
@@ -15,6 +15,7 @@ export interface IconProps extends React.HTMLAttributes<HTMLDivElement> {
   size?: IconSize;
   type?: IconType;
   title?: string;
+  containerClassName?: string;
```

Comment:
@L-M-K-B @joshhunt Not sure if this is the best way to solve this but I needed this because I had problems with vertical alignment of the icon in some places and could only solve it by changing it on the icon.

Reply:
Feel reasonably strongly against this change - className prop is one thing, but adding fooClassNames I feel is a much bigger smell...

Maybe @ashharrison90 knows a bit more about aligning icons? 😬

Reply:
The problem seems to be we use `className` already but that goes to SVG element for some reason. I would assume it would control the outer most element but I guess someone needed that. So now there is no way to control the main wrapper. The problem with vertical align seem a bit related to the context. Like in a button this may work, but for example I have a text that is written vertically instead of horizontally and there I have to tweak the alignment. Similar to my comment https://github.com/grafana/grafana/pull/69723#discussion_r1230674859 I think this is a standard use of `className` as components should not handle styling that affects it's outside positioning (like margins and maybe vertical align can be in the same group).

Reply:
A weird case in point which has both good and bad parts:
<img width="147" alt="Screenshot 2023-06-15 at 11 18 12" src="https://github.com/grafana/grafana/assets/1014802/3c8e1014-091f-460d-8eaa-e0d55c08712d">

Icon button has right margin for some reason and at the same time does not have any vertical align. So I have to use className to add vertical align (which I think is ok because I may want some specific vertical align) but I have to remove the margin which is bad imho (I should be adding margins if needed because how does the component know in what kind of layout will it be used).

Reply:
The goal we're working towards is that 'feature developers' (like yourself) can build UIs quickly and easily without having to manually write CSS themselves, as writing CSS to build consistent UIs just does not scale out to the many developers working on Grafana.

To do that, we must build/bring in better primitives (such as the recent Text component, and Stack) to solve the common cases people reach for CSS, and also fix bugs in components (like to mention). Adding `innerInnerWrapperClassName` cannot be how we fix these bugs because then everyone has to write their own css to fix the bug, it obscures missing features, and it makes the components _extremely_ difficult to maintain because now we're exposing all internal implementation details of components to consumers to fiddle with.

Reply:
Yeah I agree `innerInnerWrapperClassName` isn't ideal, but in general having the ability to style components I think is crucial. Imho thinking that having Stack will mean I don't need to set a margin or some flex attribute or alignment is untrue (and I don't consider those internals). It may work for some page layouts but not for more custom stuff. And I would rather use css than having to figure out some special props for these as I need css as a frontend developer anyway as we work with other stuff too.